controllers/awsmachine_controller.go

```diff
@@ -868,10 +869,28 @@ func (r *AWSMachineReconciler) reconcileLBAttachment(machineScope *scope.Machine
 			return r.deregisterInstanceFromClassicLB(machineScope, elbsvc, i)
 		}
 		machineScope.Debug("deregistering from v2 load balancer")
-		return r.deregisterInstanceFromV2LB(machineScope, elbsvc, i)
+		errs := []error{}
+		errs = append(errs, r.deregisterInstanceFromV2LB(machineScope, elbsvc, i, elbScope.ControlPlaneLoadBalancer()))
+		errs = append(errs, r.deregisterInstanceFromV2LB(machineScope, elbsvc, i, elbScope.SecondaryControlPlaneLoadBalancer()))
```

Comment:
Covert to a for loop?

Reply:
It might be good to add a helper in scope that returns all control plane load balancers?